lib/Sema/CSSimplify.cpp

```diff
@@ -2063,13 +2063,17 @@ static bool isInPatternMatchingContext(ConstraintLocatorBuilder locator) {
     if (path.back().is<LocatorPathElt::PatternMatch>()) {
       return true;
     } else if (path.size() > 1) {
+      auto pathElement = llvm::find_if(path, [](LocatorPathElt &elt) {
+          return elt.is<LocatorPathElt::PatternMatch>();
+      });
+      
+      if (pathElement == path.end())
+          return false;
```

Comment:
The new logic already covers the first case as well
```suggestion
      auto pathElement = llvm::find_if(path, [](LocatorPathElt &elt) {
          return elt.is<LocatorPathElt::PatternMatch>();
      });
      return pathElement != path.end();
```
So this may be enough?

Reply:
Yes👏 In my environment test seems passed.